src/cunumeric/set/unique.cu

```diff
@@ -102,13 +102,13 @@ static Piece<VAL> tree_reduce(Array& output,
       auto p_mine      = my_piece.first.ptr(0);
       auto p_other     = other_piece.first.ptr(0);
 
-      thrust::merge(thrust::cuda::par.on(stream),
+      thrust::merge(DEFAULT_POLICY.on(stream),
                     p_mine,
                     p_mine + my_piece.second,
                     p_other,
                     p_other + other_piece.second,
                     p_merged);
-      auto* end = thrust::unique(thrust::cuda::par.on(stream), p_merged, p_merged + merged_size);
+      auto* end = thrust::unique(DEFAULT_POLICY.on(stream), p_merged, p_merged + merged_size);
 
       // Make sure we release the memory so that we can reuse it
       my_piece.first.destroy();
```

Comment:
This is probably safe, because even with `par_nosync` the `thrust::unique` operation must internally sync, so it can have a valid `end` pointer before returning. It would be good to note this.